### PR TITLE
fix(test): Use RuleChain to operate rules sequentially

### PR DESCRIPTION
**Pull request by djansen-redhat**
- Change BeforeClass to Before
- Make HasEmailRules BeforeClass
- Fixed a couple of random weak points (hasNotification > expectNotification)
- Put a waitForPageSilence after clicking a text field, in case it blurs

### CODE REVIEW

**Comment by seanf** [comment]
@djansen-redhat It would be good to get this into release, when it's working.

**Comment by seanf** [comment]
:+1: in principle for more consistent tests

:-1: in practice for breaking all the tests (consistently?) :-(

**Comment by djansen-redhat** [comment]
@seanf Hah, yeah... I submitted it as a WIP to see what the damage was.

File: functional-test/src/test/java/org/zanata/feature/account/InactiveUserLoginTest.java [hunk 1]
```diff
@@ -108,8 +99,7 @@ public void resendActivationEmail() throws Exception {
                 .signInInactive(usernamepassword, usernamepassword)
                 .clickResendActivationEmail();
 
-        assertThat(homePage.getNotificationMessage())
-                .isEqualTo(HomePage.SIGNUP_SUCCESS_MESSAGE)
+        assertThat(homePage.expectNotification(HomePage.SIGNUP_SUCCESS_MESSAGE))
```
Comment (djansen-redhat) [inline functional-test/src/test/java/org/zanata/feature/account/InactiveUserLoginTest.java]: Fixed intermittent breakage here.

**Comment by djansen-redhat** [comment]
The "import fixing" feature of Intellij seems to be quite aggressive. Why it changes some to name.* is beyond me.

**Comment by djansen-redhat** [comment]
@seanf any further comment on this one?

File: functional-test/src/test/java/org/zanata/feature/administration/EditTranslationMemoryTest.java [hunk 2]
```diff
@@ -47,13 +45,10 @@
...
 
-    @ClassRule
-    public static AddUsersRule addUsersRule = new AddUsersRule();
-
     private TestFileGenerator testFileGenerator = new TestFileGenerator();
 
-    @BeforeClass
-    public static void beforeClass() {
+    @Before
+    public void beforeClass() {
```
Comment (seanf) [inline functional-test/src/test/java/org/zanata/feature/administration/EditTranslationMemoryTest.java]: Should rename to `before()`.

**Comment by seanf** [comment]
:+1: Good cleanup!

File: functional-test/src/test/java/org/zanata/feature/administration/EditTranslationMemoryTest.java (hunk 2 above)
Comment (djansen-redhat) [inline functional-test/src/test/java/org/zanata/feature/administration/EditTranslationMemoryTest.java]: Damn, nice catch.  Fixed and pushed.

**Comment by seanf** [comment]
@djansen-redhat LGTM

**Comment by djansen-redhat** [comment]
Had to Google that one...

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java [hunk 3]
```diff
@@ -400,8 +400,12 @@ public void enterText(final WebElement element, final String text, boolean clear
         } else {
             Actions enterTextAction = new Actions(getDriver()).moveToElement(element);
             enterTextAction = enterTextAction.click();
+            // Fields can 'blur' on click
+            waitForPageSilence();
             if (clear) {
                 enterTextAction = enterTextAction.sendKeys(Keys.chord(Keys.CONTROL, "a")).sendKeys(Keys.DELETE);
+                // Fields can 'blur' on clear
+                waitForPageSilence();
```
Comment (djansen-redhat) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: @seanf Added this after you set "Reviewed"

**Pull request merged**